service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java

```diff
@@ -0,0 +1,49 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import static org.hl7.fhir.dstu3.model.Bundle.BundleType.TRANSACTION;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.EpisodeOfCare;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+
+@Component
+public class EncounterReportBundleService {
+
+    @Autowired
+    private EncounterMapper encounterMapper;
+
+    @Autowired
+    private EpisodeOfCareMapper episodeOfCareMapper;
+
+    public Bundle createEncounterBundle(POCDMT000002UK01ClinicalDocument1 clinicalDocumentDocument) {
+        Bundle bundle = new Bundle();
+        bundle.setType(TRANSACTION);
+
+        Encounter encounter = encounterMapper.mapEncounter();
+
+        addEncounter(bundle, encounter);
+        addEpisodeOfCare(bundle, encounter);
+
+        return bundle;
+    }
+
+    private void addEncounter(Bundle bundle, Encounter encounter) {
+        bundle.addEntry()
+            .setFullUrl(encounter.getIdElement().getValue())
```

Comment:
IdElement is random UUID -> why is it set here as url?

Reply:
URL is a "resource locator" so UUID identifies a resource quite well within a bundle message.
See: https://hapifhir.io/hapi-fhir/docs/client/examples.html

Reply:
Good idea. Done.